sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -174,11 +167,8 @@ export class IER12FileDetail implements IER12FileLine {
     record.appendFormattedDate(this.studentBirthDate);
     record.appendStringWithFiller(this.studentGroupCode, 4);
     record.appendStringWithFiller(this.studentMaritalStatusCode, 4);
-    record.appendOptionalStringWithFiller(this.studentDisabilityStatusCode, 4);
-    record.appendOptionalStringWithFiller(
-      this.applicationDisabilityStatusFlag,
-      1,
-    );
+    record.append(this.studentDisabilityStatusCode);
+    record.append(this.applicationDisabilityStatusFlag);
```

Comment:
These seem to be the only fields in which a wrong input (anything outside this class) could derail all the other indexes.
Using the `record.appendStringWithFiller` instead of `record.append` would prevent a bigger issue if, for instance, the `studentDisabilityStatusCode` is provided with something different than 4 caracters.

Reply:
I initially thought that, the transformation which produce the value of these fields will have exactly the same length as expected always. But now I see the point that, if some bug in introduced in these transformations then it will be very had to debug, if one or more fields derail the index. I can change it.